src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -344,12 +374,20 @@ void OnPlatformEventHandler(const chip::DeviceLayer::ChipDeviceEvent * event, in
 // As per specifications section 11.22.5.1. Constant RESP_MAX
 constexpr size_t kMaxRspLen = 900;
 
+// TODO: The code currently has two sources of truths for fabrics, the fabricInfo table + the attributes. There should only be one,
+// the attributes list. Currently the attributes are not persisted so we are keeping the fabric table to have the
+// fabrics/admrins be persisted. Once attributes are persisted, there should only be one sorce of truth, the attributes list and
+// only that should be modifed to perosst/read/write fabrics.
+// TODO: Once attributes are persisted, implement reading/writing/manipulation fabrics around that and remove fabricTable
+// logic.
 class OpCredsFabricTableDelegate : public chip::FabricTable::Delegate
 {
 public:
-    // Gets called when a fabric is about to be deleted
-    void FabricWillBeRemoved(const FabricTable & fabricTable, FabricIndex fabricIndex) override
```

Comment:
Why was it changed? The events must be emitted before removing a fabric and its ACLs. It has been broken multiple times in the past :)